ansible_collections/arista/avd/roles/eos_designs/python_modules/base/avdstructuredconfig.py

```
@@ -320,8 +328,24 @@ def snmp_server(self):
             return None
 
         snmp_server = {}
-        if (compute_local_engineid := snmp_settings.get("compute_local_engineid")) is True:
-            local_engine_id = sha1(f"{self._hostname}{self._mgmt_ip}".encode("utf-8")).hexdigest()
+
+        local_engine_id = None
+
+        if snmp_settings.get("compute_local_engineid") is True:
+            compute_source = get(snmp_settings, "compute_local_engineid_source", default="hostname_and_ip")
+            if compute_source == "hostname_and_ip":
+                local_engine_id = sha1(f"{self._hostname}{self._mgmt_ip}".encode("utf-8")).hexdigest()
+            elif compute_source == "system_mac":
+                if self._system_mac_address is None:
+                    raise AristaAvdError("default_engine_id_from_system_mac: true requires system_mac_address to be set!")
```

This should raise the missingvariable error.

---

Fixed in 4bcf83d